metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/TerraMetricsSubmitter.java

```diff
@@ -0,0 +1,572 @@
+                        Map<String, List<CSVRecord>> sourceUrlToCsvRecords = workflowMetricsToProcess.stream()
+                                .collect(groupingBy(csvRecord -> csvRecord.get(TerraMetricsCsvHeaders.source_url)));
+                        /*
+                        for (CSVRecord workflowMetricToProcess: workflowMetricsToProcess) {
+                            final String csvRecordSourceUrl = workflowMetricToProcess.get(TerraMetricsCsvHeaders.source_url);
+                            List<CSVRecord> csvRecordsWithSameSourceUrl = sourceUrlToCsvRecords.getOrDefault(csvRecordSourceUrl, new ArrayList<>());
+                            csvRecordsWithSameSourceUrl.add(workflowMetricToProcess);
+                            sourceUrlToCsvRecords.put(csvRecordSourceUrl, csvRecordsWithSameSourceUrl);
+                        }
+                         */
+
+                        LOG.info("Processing rows {} to {}", workflowMetricsToProcess.get(0).getRecordNumber(), workflowMetricsToProcess.get(workflowMetricsToProcess.size() - 1).getRecordNumber());
+
+                        sourceUrlToCsvRecords.entrySet().stream()
+                                .parallel()
+                                .forEach(entry -> {
+                                    final String sourceUrl = entry.getKey();
+                                    final List<CSVRecord> csvRecordsWithSameSourceUrl = entry.getValue();
+                                    submitWorkflowExecutions(sourceUrl, csvRecordsWithSameSourceUrl, workflowsApi, extendedGa4GhApi, skippedExecutionsCsvPrinter);
+                                });
+
+                        workflowMetricsToProcess.clear();
+                        logStats();
+                    }
+                }
+            } catch (IOException e) {
+                LOG.error("Unable to create new CSV output file", e);
+                System.exit(FAILURE_EXIT_CODE);
+            }
+
+            logStats();
+
+            if (submitTerraMetricsCommand.isRecordSkippedExecutions()) {
+                LOG.info("View skipped executions in file {}", outputFileName);
+            }
+        } catch (IOException e) {
+            LOG.error("Unable to read input CSV file", e);
+            System.exit(FAILURE_EXIT_CODE);
+        }
+    }
+
+    private void logStats() {
+        LOG.info("Done processing {} executions from Terra. Submitted {} executions. Skipped {} executions.", numberOfExecutionsProcessed, numberOfExecutionsSubmitted, numberOfExecutionsSkipped);
+    }
+
+    /**
+     * Processes the workflow execution from the CSV record and submits the execution to Dockstore if it is valid.
+     * @param workflowMetricRecord
+     * @param workflowsApi
+     * @param extendedGa4GhApi
+     * @param skippedExecutionsCsvPrinter
+     */
+    private void processWorkflowExecution(CSVRecord workflowMetricRecord, WorkflowsApi workflowsApi, ExtendedGa4GhApi extendedGa4GhApi, CSVPrinter skippedExecutionsCsvPrinter) {
```

Comment:
This method not used? At least the IDE claims it's not, unless the caches are corrupt or something.

Reply:
Your IDE was right, it wasn't used so I removed it